drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb

```diff
@@ -206,29 +206,55 @@ def load_source_file_pg(read_from:, klass:)
       # logger.debug { "   #{copy_sql}" }
 
       lines_loaded = nil
+      total_lines = nil
+      expect_col_count = copy_cols.size - meta_data.size
+      row_errors = []
       # SLOW_CHECK; klass.connection.transaction do
       bm = Benchmark.measure do
         pg_conn = klass.connection.raw_connection
         pg_result = pg_conn.copy_data copy_sql do
           read_from.rewind
-          CSV.parse(read_from, headers: false, liberal_parsing: true) do |row|
-            # ensure the row is long enough before
-            # we put the typed meta_data at the end
-            # all loader tables should handle nil data for columns
-            # with missing values
-            row << nil while row.size < copy_cols.size - meta_data.size
-            pg_conn.put_copy_data (row + meta_data).to_csv
-
-            # SLOW_CHECK; data = copy_cols.zip(row + meta_data).to_h
-            # SLOW_CHECK;  klass.create! data
+          begin
+            parser = CSV.new(read_from, headers: false, liberal_parsing: true)
+            parser.each do |row|
+              # There were excess columns, probably due to an unquoted comma
+              if row.size > expect_col_count
+                row_errors << {
+                  file_name: base_name,
+                  message: 'Extra columns found',
+                  details: "lineno: #{parser.lineno}",
```

Comment:
Any reason we can't put "Line number: ..." here?

Reply:
Nope... will change